SimulationIOFilters/Utility/OnScaleTableFileWriter.h

```
@@ -1,14 +1,150 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+ * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The code contained herein was partially funded by the followig contracts:
+ *    United States Air Force Prime Contract FA8650-15-D-5231
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ */
+
 #pragma once
 
 #include <QtCore/QString>
+#include <QtCore/QDir>
+#include <QtCore/QFile>
+#include <QtCore/QTextStream>
 
 #include "SIMPLib/Common/SIMPLArray.hpp"
 #include "SIMPLib/DataArrays/DataArray.hpp"
 #include "SIMPLib/DataArrays/StringDataArray.h"
 #include "SIMPLib/Geometry/ImageGeom.h"
 
+#include "SimulationIO/SimulationIOFilters/Utility/EntriesHelper.h"
+
 namespace OnScaleTableFileWriter
 {
-bool write(const ImageGeom& imageGeom, const StringDataArray& phaseNames, const DataArray<int32_t>& featureIds, const QString& outputPath, const QString& filePrefix,
-           const IntVec3Type& numKeypoints);
+void writeCoords(QTextStream& stream, const QString& text, size_t maxIndex, float origin, float spacing);
+
+/**
+ * @brief Writes a OnScale table file. All elements of featureIds must be >=0. Returns true on success.
+ * @param imageGeom
+ * @param phaseNames
+ * @param featureIds
+ * @param outputPath
+ * @param filePrefix
+ * @param numKeypoints
+ * @return
+ */
+template <class T>
+bool write(const ImageGeom& imageGeom, const StringDataArray& phaseNames, const DataArray<T>& featureIds, const QString& outputPath, const QString& filePrefix, const IntVec3Type& numKeypoints)
+{
```

You may want to consider how to pass back an error code & error string so if something happens we can report the error as a human readable error code.